### PR TITLE
Improve http request

### PR DESCRIPTION
**Pull request by nwlandry**
* Added an if-else statement to throw an intelligible error if the `load_xgi_data` request is not successful
* Added the ability to cache the results of querying XGI-DATA.
* Changed the default to `read=False`.
* Updated contributors.

### CODE REVIEW

File: xgi/readwrite/xgi_data.py [hunk 1]
```diff
@@ -101,14 +114,46 @@ def _request_from_xgi_data(dataset):
     """
 
     index_url = "https://gitlab.com/complexgroupinteractions/xgi-data/-/raw/main/index.json?inline=false"
-    index = requests.get(index_url).json()
+    r = requests.get(index_url)
+
+    if r.ok:
+        index_data = r.json()
+    else:
+        raise XGIError("Error: HTTP response {r.status_code}")
```
Comment (maximelucas) [inline xgi/readwrite/xgi_data.py:122]:
```suggestion
        raise XGIError(f"Error: HTTP response {r.status_code}")
```

File: xgi/readwrite/xgi_data.py [hunk 2]
```diff
@@ -101,14 +114,46 @@ def _request_from_xgi_data(dataset):
...
-        print(*index, sep="\n")
+        print(*index_data, sep="\n")
         raise XGIError("Must choose a valid dataset name!")
 
-    data = requests.get(index[key]["url"]).json()
+    r = requests.get(index_data[key]["url"])
+    if r.ok:
+        return r.json()
+    else:
+        raise XGIError("Error: HTTP response {r.status_code}")
```
Comment (maximelucas) [inline xgi/readwrite/xgi_data.py:134]:
```suggestion
        raise XGIError(f"Error: HTTP response {r.status_code}")
```

File: xgi/readwrite/xgi_data.py (hunk 1 above)
Comment (lordgrilo) [reply xgi/readwrite/xgi_data.py:122]: +1

File: xgi/readwrite/xgi_data.py (hunk 2 above)
Comment (lordgrilo) [reply xgi/readwrite/xgi_data.py:134]: +1

**Review by lordgrilo — APPROVED** [review]
I support all the comments by @maximelucas and I approve the merge

File: xgi/readwrite/xgi_data.py (hunk 1 above)
Comment (nwlandry) [reply xgi/readwrite/xgi_data.py]: Done

File: xgi/readwrite/xgi_data.py (hunk 2 above)
Comment (nwlandry) [reply xgi/readwrite/xgi_data.py]: done

**Pull request merged**